packages/engine/src/avatar/systems/AvatarTransparencySystem.tsx

```diff
@@ -54,34 +57,37 @@ import { AvatarHeadDecapComponent } from '../components/AvatarIKComponents'
 
 const headDithering = 0
 const cameraDithering = 1
-
+const avatarQuery = defineQuery([AvatarComponent])
 const execute = () => {
   const selfEntity = AvatarComponent.getSelfAvatarEntity()
   if (!selfEntity) return
-  const materials = getComponent(selfEntity, TransparencyDitheringRoot)?.materials
-  if (!materials) setComponent(selfEntity, TransparencyDitheringRoot, { materials: [] })
-
-  const cameraAttached = getState(XRControlsState).isCameraAttachedToAvatar
-  const avatarComponent = getComponent(selfEntity, AvatarComponent)
-  const cameraComponent = getOptionalComponent(Engine.instance.viewerEntity, FollowCameraComponent)
-
-  if (!materials?.length) return
-  for (const materialUUID of materials) {
-    const pluginComponent = getOptionalComponent(
-      UUIDComponent.getEntityByUUID(materialUUID),
-      TransparencyDitheringPlugin
-    )
-    if (!pluginComponent) continue
-    pluginComponent.centers.value[headDithering].setY(avatarComponent.eyeHeight)
-    pluginComponent.distances.value[headDithering] =
-      cameraComponent && !cameraAttached ? Math.max(Math.pow(cameraComponent.distance * 5, 2.5), 3) : 3.25
-    pluginComponent.exponents.value[headDithering] = cameraAttached ? 12 : 8
-    pluginComponent.useWorldCalculation.value[headDithering] = ditherCalculationType.localPosition
-    const viewerPosition = getComponent(Engine.instance.viewerEntity, TransformComponent).position
-    pluginComponent.centers.value[cameraDithering].set(viewerPosition.x, viewerPosition.y, viewerPosition.z)
-    pluginComponent.distances.value[cameraDithering] = cameraAttached ? 8 : 3
-    pluginComponent.exponents.value[cameraDithering] = cameraAttached ? 10 : 2
-    pluginComponent.useWorldCalculation.value[cameraDithering] = ditherCalculationType.worldTransformed
+  for (const entity of avatarQuery()) {
+    const materials = getComponent(entity, TransparencyDitheringRoot)?.materials
+    if (!materials) setComponent(entity, TransparencyDitheringRoot, { materials: [] })
+
+    const cameraAttached = getState(XRControlsState).isCameraAttachedToAvatar
+    const avatarComponent = getComponent(entity, AvatarComponent)
+    const cameraComponent = getOptionalComponent(getState(EngineState).viewerEntity, FollowCameraComponent)
+
+    if (!materials?.length) return
+    for (const materialUUID of materials) {
+      const pluginComponent = getOptionalComponent(
+        UUIDComponent.getEntityByUUID(materialUUID),
+        TransparencyDitheringPlugin
+      )
+      if (!pluginComponent) continue
+      const viewerPosition = getComponent(Engine.instance.viewerEntity, TransformComponent).position
```

Comment:
Super small, but if this viewerEntity is the same as the one being retrieved from the EngineState on line 70 can it be reused here? Otherwise looks great, approved

Reply:
Thanks Michael, good call out. Will do.